### PR TITLE
implementing errors spi

### PR DESCRIPTION
**Pull request by michaelir**

### CODE REVIEW

File: libs/velo-external-db-commons/src/libs/errors.ts [hunk 1]
```diff
@@ -1,90 +1,106 @@
...
+    constructor(message: string) {
         super(message)
-        this.status = status
     }
 }
 
 export class UnauthorizedError extends BaseHttpError {
     constructor(message: string) {
-        super(message, 401)
+        super(message)
```
Comment (Idokah) [inline libs/velo-external-db-commons/src/libs/errors.ts:9]: Why did you remove all the status codes?

Comment (michaelir) [reply libs/velo-external-db-commons/src/libs/errors.ts:9]: Since they are now irrelevant. The status code is determined via errors.ts

File: apps/velo-external-db/test/e2e/app_data.e2e.spec.ts [hunk 2]
```diff
@@ -296,6 +297,22 @@ describe(`Velo External DB Data REST API: ${currentDbImplementationName()}`,  ()
...
 
+    test('count api on non existing collection should fail with 404', async() => {
+        await schema.givenCollection(ctx.collectionName, [ctx.column], authOwner)
+        await givenItems([ctx.item, ctx.anotherItem], ctx.collectionName, authAdmin)
+        await expect( 
+            axiosInstance.post('/data/count', countRequest(gen.randomCollectionName()), authAdmin) 
+        ).rejects.toThrow('404')
+    })
+
+    test.only('insert api on non existing collection should fail with 404', async() => {
```
Comment (Idokah) [inline apps/velo-external-db/test/e2e/app_data.e2e.spec.ts:308]: remove .only

File: apps/velo-external-db/test/e2e/app_schema_hooks.e2e.spec.ts [hunk 3]
```diff
@@ -139,7 +139,7 @@ describe(`Velo External DB Schema Hooks: ${currentDbImplementationName()}`, () =
                     })
 
                     await expect(axios.post('/schemas/column/remove', { collectionName: ctx.collectionName, columnName: ctx.column.name }, authOwner)).rejects.toMatchObject(
-                        errorResponseWith(400, 'Should not be removed')
+                        errorResponseWith(500, 'Should not be removed')
```
Comment (Idokah) [inline apps/velo-external-db/test/e2e/app_schema_hooks.e2e.spec.ts:142]: the SPI expects here 500? (and in all the others that you've changed in this file ? )
thats weird because it's clearly not 500, maybe we should tell them.

BTW, the test name is 'should throw 400' so if you decide it's really 500 please change the test names.

File: libs/external-db-mysql/src/mysql_data_provider.ts [hunk 4]
```diff
@@ -45,11 +45,13 @@ export default class DataProvider implements IDataProvider {
         
         const data = items.map((item: Item) => asParamArrays( patchItem(item) ) )
         const resultset = await this.query(sql, [data])
-                                    .catch( translateErrorCodes )
+                                    .catch( err => translateErrorCodes(err, collectionName) )
         return resultset.affectedRows
     }
 
     async update(collectionName: string, items: Item[]): Promise<number> {
+        console.log({items})
```
Comment (Idokah) [inline libs/external-db-mysql/src/mysql_data_provider.ts:53]: remove logs

File: libs/velo-external-db-core/src/web/error-middleware.ts [hunk 5]
```diff
@@ -1,9 +1,14 @@
...
 
 export const errorMiddleware = (err: any, _req: any, res: Response, _next?: NextFunction) => {
   if (process.env['NODE_ENV'] !== 'test') {
     console.error(err)
   }
-  res.status(err.status || 500)
-     .send({ message: err.message })
+
+
+  console.log({err})
```
Comment (Idokah) [inline libs/velo-external-db-core/src/web/error-middleware.ts:10]: can remove log

File: libs/velo-external-db-core/src/web/domain-to-spi-error-translator.ts [hunk 6]
```diff
@@ -0,0 +1,25 @@
+import { errors as domainErrors} from '@wix-velo/velo-external-db-commons'
```
Comment (Idokah) [inline libs/velo-external-db-core/src/web/domain-to-spi-error-translator.ts:1]: IMO we create unit tests on that function (even though it is pretty straightforward), just as we do for other translators.

**Comment by Idokah** [comment]
BTW you have conflicts

**Review by Idokah — APPROVED** [review]
